cibuildwheel/docker_container.py

```diff
@@ -57,6 +57,9 @@ def __enter__(self) -> 'DockerContainer':
             ],
             stdin=subprocess.PIPE,
             stdout=subprocess.PIPE,
+            # output the subcommand stderr to our stdout. Avoids ordering
+            # problems between log messages on stderr and stdout
+            stderr=sys.stdout,
```

Comment:
What about users piping stdout/stderr separately, and expecting stderr to contain text if things go wrong? Does that happen? Is that something to worry about?

Reply:
Yeah, so this change was motivated by Github Actions, where some output on stderr would end up on the wrong side of the fold group and break the formatting. 

> What about users piping stdout/stderr separately, and expecting stderr to contain text if things go wrong? Does that happen? Is that something to worry about?

The output of cibuildwheel isn't intended to be machine readable at all, so I don't expect users to pipe the output of cibuildwheel to anywhere, really. Maybe to their own log file? either way, I see keeping output interleaved as more important than the separate streams. (Also, in tests, the way the pytest splits them up makes it harder to follow execution when something breaks)

Reply:
Fair enough, yes! One thing I just thought of: we do have `sys.stdout = Unbuffered(sys.stdout)  # type: ignore`. Would it be worth trying to add the same for `sys.stderr`?

Apart from that, yes, I think you're right. That's more important indeed.

Reply:
I think stderr is unbuffered by default, but it's a good idea, I'll give it a shot. Weirdly it only seems to be a problem on Github Actions.

Reply:
Hm! So I tried this, and removing the stdout-stderr redirect, and saw no problem! Then I tried it without any workarounds, and it still worked fine. So I guess I'll leave it without any workarounds for now, if we see output bleeding out of the fold groups we can add it back. :)

Reply:
Huh, OK, then. I'll keep this in the back of mind as well! If I (we) catch things getting mixed up gaain, we can get back to it!
Thanks for trying! :-)